### PR TITLE
For #17084: Use in-app browser switching dialog

### PR DESCRIPTION
**Pull request by hakkikaancaliskan**
https://user-images.githubusercontent.com/17825767/103143382-e36f0d00-4725-11eb-9387-d00d731502ee.mp4

I'm unable to differentiate if user selected "don't ask again" before or selected another browser. If the current behaviour okey, then it's fine.
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:433]: Can you add a description to this function explaining the scenarios for each version? Looks like we specify the listener for Q and N

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:478]: I like this comment, but I think we can just keep the first sentence and take out "Possible reasons"

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:485]: How are we handling other versions?

**Review by eliserichards — CHANGES_REQUESTED** [review]
This looks good so far! I have a few nits 😄

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:485]: For >Q -> open in-app browser switching dialog
For >N -> open android settings and navigate to default apps settings
For <N -> Load sumo page to show how to change default app

**Comment by hakkikaancaliskan** [comment]
@eliserichards Adressed changes and squashed commits, could you please review again?

**Review by eliserichards — APPROVED** [review]
Awesome! Thank you so much for addressing everything @hakkikaancaliskan! 🎉

**Comment by eliserichards** [comment]
bors try

**Comment by hakkikaancaliskan** [comment]
ping @eliserichards

**Comment by eliserichards** [comment]
Sorry @hakkikaancaliskan 😉 👍

**Comment by hakkikaancaliskan** [comment]
@eliserichards when you cherry pick commit in other pr, this pr also runs ci. So you don't need to wait for other pr to get approval 😄

**Pull request closed without merging**
